tests/e2e/specs/auth-flow.test.js

```diff
@@ -0,0 +1,71 @@
+/* eslint-env node */
+/* global page, jestPuppeteer */
+
+/**
+ * WordPress dependencies
+ */
+import { activatePlugin, deactivatePlugin, createURL, visitAdminPage } from '@wordpress/e2e-test-utils';
+
+/**
+ * Internal dependencies
+ */
+import { deactivateAllOtherPlugins, resetSiteKit } from '../utils';
+
+const oauthClientConfig = JSON.stringify( {
+	'web': {
+		'client_id': 'test-client-id',
+		'client_secret': 'test-client-secret',
+		'project_id': 'test-project-id',
+		'auth_uri': 'https://accounts.google.com/o/oauth2/auth',
+		'token_uri': 'https://accounts.google.com/o/oauth2/token',
+		'auth_provider_x509_cert_url': 'https://www.googleapis.com/oauth2/v1/certs'
+	}
+} );
+
+function stubGoogleSignIn( request ) {
+	if ( request.url().startsWith( 'https://accounts.google.com/o/oauth2/auth' ) ) {
+		request.respond( {
+			status: 302,
+			headers: {
+				location: createURL( '/', 'oauth2callback=1&code=valid-test-code' )
+			}
+		} );
+	} else {
+		request.continue();
+	}
+}
+
+describe( 'Site Kit set up flow for the first time', () => {
+
+	beforeAll( async() => {
+		await deactivateAllOtherPlugins();
```

Comment:
Can you explain the rationale behind this? Why is this needed?

If you're running into issues otherwise we'd need verify whether this happens for other tests as well. If so, we could automate this.

Reply:
The main reason is that core's `deactivatePlugin` will fail if the plugin is already deactivated (unlike `activatePlugin`). This test uses multiple utilities so this is an idempotent way to ensure that they're all deactivated without needing to check if they're active or not. It makes more sense in the context of a local environment where we're running the same tests over and over again, and different plugins could be still active or not from the previous run but I think it's a reasonable utility to have.

Reply:
> The main reason is that core's `deactivatePlugin` will fail if the plugin is already deactivated

Sounds like something that should be reported upstream 🙂 

See https://github.com/WordPress/gutenberg/pull/16816

> It makes more sense in the context of a local environment where we're running the same tests over and over again, and different plugins could be still active or not from the previous run but I think it's a reasonable utility to have.

Another thing to put into `bootstrap.js`?

To clarify, you're not running the tests in the isolated Docker container?

Reply:
> Sounds like something that should be reported upstream 🙂

Thanks! 😄 

> Another thing to put into bootstrap.js?

Yeah, it seems like that would be useful as well. That way, each test would start knowing only Site Kit is active.

> To clarify, you're not running the tests in the isolated Docker container?

I am running them in the Docker container, but it doesn't fully reset the state on each run and it would be too inefficient to run `env:reset-site` every time.